rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -282,6 +360,92 @@ trait NodeInfoServiceCached extends NodeInfoService  with Loggable with CachedRe
   }
 
   /**
+   * An utility method that gets data from backend for things that are
+   * node really nodes (pending or deleted).
+   */
+  private[this] def getNotAcceptedNodeDataFromBackend(status: InventoryStatus): Box[Map[NodeId, NodeInfo]] = {
+    import scala.collection.mutable.{Map => MutMap}
+
+    //some map of things - mutable, yes
+    val nodeInventories = MutMap[String, LDAPEntry]() // node_uuid -> entry
+    val machineInventories = MutMap[String, LDAPEntry]() // machine_dn -> entry
+
+    ldap.map { con =>
+
+      val allEntries = getNodeInfoEntries(con, searchAttributes, status)
+      // now, create the nodeInfo
+      allEntries.foreach { e =>
+        if(e.isA(OC_MACHINE)) {
+          machineInventories += (e.dn.toString -> e)
+        } else if(e.isA(OC_NODE)) {
+          nodeInventories += (e.value_!(A_NODE_UUID) -> e)
+        } else {
+          // it's an error, don't use
+        }
+      }
+
+      nodeInventories.flatMap { case (id, nodeEntry) =>
+        val machineInfo = for {
+                            containerDn  <- nodeEntry(A_CONTAINER_DN)
+                            machineEntry <- machineInventories.get(containerDn)
+                          } yield {
+                            machineEntry
+                          }
+          for {
+          nodeInfo <- ldapMapper.convertEntriesToSpecialNodeInfos(nodeEntry, machineInfo)
+        } yield {
+          (nodeInfo.id, nodeInfo)
+        }
+      }.toMap
+    }
+  }
+
+  override final def getPendingNodeInfos(): Box[Map[NodeId, NodeInfo]] = getNotAcceptedNodeDataFromBackend(PendingInventory)
+
+  override final def getDeletedNodeInfos(): Box[Map[NodeId, NodeInfo]] = getNotAcceptedNodeDataFromBackend(RemovedInventory)
+
+  private[this] def getNotAcceptedNodeInfo(nodeId: NodeId, status: InventoryStatus): Box[Option[NodeInfo]] = {
+    val dit = status match {
+      case AcceptedInventory => inventoryDit
+      case PendingInventory  => pendingDit
+      case RemovedInventory  => removedDit
+    }
+
+    def sureGet(con: RoLDAPConnection, dn: DN): Box[Option[LDAPEntry]] = {
```

Comment:
it's a method to factor the common code for not accepted (i.e, deleted or pending) nodes